### ISSUE TITLE
0.18.3 regresion on github-workflows schema

### ISSUE THREAD

**Issue by ssbarnea** [python-jsonschema/check-jsonschema#157]
0.18.3 includes changes to github-workflow schema that makes is report failure when it is not supposed to fail. See https://results.pre-commit.ci/run/github/12106931/1664817429.-Cx_j6HYRsWF_lHn12MIyg

```
Schema validation errors were encountered.
  .github/workflows/tox.yml::$.jobs.linters: {'name': '${{ matrix.env.TOXENV }}', 'runs-on': '${{ matrix.os }}', 'strategy': {'fail-fast': False, 'matrix': {'python-version': [3.9], 'os': ['ubuntu-20.04'], 'env': [{'TOXENV': 'lint'}, {'TOXENV': 'docs'}, {'TOXENV': 'eco'}, {'TOXENV': 'packaging'}, {'TOXENV': 'validate-hook'}]}}, 'steps': [{'name': 'Check out src from Git', 'uses': 'actions/checkout@v3', 'with': {'fetch-depth': 0}}, {'name': 'Set up Python ${{ matrix.python-version }}', 'uses': 'actions/setup-python@v4', 'with': {'python-version': '${{ matrix.python-version }}', 'cache': 'pip'}}, {'name': 'Install tox', 'run': 'python3 -m pip install --upgrade pip\npython3 -m pip install --upgrade tox\n'}, {'name': 'Log installed dists', 'run': 'python -m pip freeze --all'}, {'name': 'Initialize tox envs', 'run': 'python -m tox --parallel auto --parallel-live --notest --skip-missing-interpreters false -vv', 'env': '${{ matrix.env }}'}, {'name': 'Test with tox', 'run': 'python -m tox --parallel auto --parallel-live\n', 'env': '${{ matrix.env }}'}, {'name': 'Archive logs', 'uses': 'actions/upload-artifact@v3', 'with': {'name': 'logs.zip', 'path': '.tox/**/log/'}}]} is not valid under any of the given schemas
  Underlying errors caused this.
  Best Match:
    $.jobs.linters: 'uses' is a required property
```

I mention that the schema is valid and that uses is not needed in all cases. In addition to this, I know that current version of the schema from https://json.schemastore.org/github-workflow.json does validate correctly with other validators. I am not sure if a bad or outdated schema was used instead or the validator hit a bug, but the regression is obvious between 0.18.2 and 0.18.3

**Comment by sirosen** [comment]
Thanks for the bug report! I think there's been a (significant) regression between
https://github.com/SchemaStore/schemastore/pull/2461
and
https://github.com/SchemaStore/schemastore/pull/2465

If you ask for verbose output, you'll see all of the errors for your case. By looking up the ansible-lint jobs, I was able to find some failing workflow data where `env` was failing to validate due to these changes. In the future, it would help if you could provide a repro sample in the issue if you can; it makes working on fixes much easier.

> In addition to this, I know that current version of the schema from https://json.schemastore.org/github-workflow.json does validate correctly with other validators.

Are you certain about this? I think the issue is that `env` is no longer allowed to hold an arbitrary expression-syntax block. I'm working on a fix for schemastore right now, along with a test case to catch this.

**Comment by ssbarnea** [comment]
Ohh, sorry for not looking deeper. It is bit weird because my vscode-yaml did not indicate any error with the schema and it is always using the latest schema (checks it in background). Only check-jsonschema failed at that time.

Still, there is a slight chance that different validators would behave differently.

Closing as we will track https://github.com/SchemaStore/schemastore/pull/2510
